Cluster-scope-like behavior, autoinstall of KnativeEventingKafka as knative-eventing/knative-eventing-kafka

syedriko opened the pull request.

syedriko commented:
A port of https://github.com/openshift-knative/knative-eventing-operator/pull/45 to the Kafka operator

syedriko commented:
Do not merge just yet

matzew commented on deploy/olm-catalog/knative-kafka-operator/0.6.0/knative-kafka-operator.v0.6.0.clusterserviceversion.yaml:
```diff
@@ -156,9 +156,9 @@ spec:
+  - supported: false
     type: OwnNamespace
-  - supported: true
+  - supported: false
```
this means, you need to update the PR 420 on the community-operator repo :-)

syedriko replied:
Absolutely, as soon as this PR merges

matzew commented:
@syedriko let's add something along the lines of https://github.com/openshift-knative/knative-eventing-operator/commit/d6c59b4b237a1f616527fd6bb31e6796ef712823   to give some clarification to our users

matzew commented:
If @jcrossley3 has no concerns - :shipit:

jcrossley3 commented on pkg/controller/knativeeventingkafka/knativeeventingkafka_controller.go:
```diff
@@ -23,6 +24,11 @@ import (
 )
 
+const (
+	ns = "knative-eventing"
```
Sharing a ns with another component _may_ be a problem. But if it is, I'd prefer to fix manifestival than force you to use your own ns. Just be aware that if an operator creates a ns, it will probably try to delete that ns when the associated CR is deleted. If the ns was already there, it won't.

syedriko replied:
This latest iteration of the Kafka operator places all the ns-scoped k8s objects it creates into the knative-eventing namespace, which the code assumes to exist, since the Kafka objects depend on Eventing being installed [in the knative-eventing namespace]. So I think we're in the clear wrt to the Kafka operator trying to remove the knative-eventing namespace.

syedriko commented:
> Why so many vendor changes on this?

That's what `dep ensure -update` pulled in...

jcrossley3 commented:
> > Why so many vendor changes on this?
> 
> That's what `dep ensure -update` pulled in...

You gotta be careful with that. I typically only use that to update a specific dep. In the future, whenever you feel the need to update *everything*, split that out into its own commit/pr, just to facilitate rolling back if need be.

syedriko commented:
> > > Why so many vendor changes on this?
> > 
> > 
> > That's what `dep ensure -update` pulled in...
> 
> You gotta be careful with that. I typically only use that to update a specific dep. In the future, whenever you feel the need to update _everything_, split that out into its own commit/pr, just to facilitate rolling back if need be.

Fixed, only manifestival is updated.

syedriko commented:
> @syedriko let's add something along the lines of [openshift-knative/knative-eventing-operator@d6c59b4](https://github.com/openshift-knative/knative-eventing-operator/commit/d6c59b4b237a1f616527fd6bb31e6796ef712823) to give some clarification to our users

Done
